registration/include/pcl/registration/impl/ndt.hpp

```diff
@@ -95,26 +95,23 @@ NormalDistributionsTransform<PointSource, PointTarget>::computeTransformation (P
   }
 
   // Initialize Point Gradient and Hessian
-  point_gradient_.setZero ();
-  point_gradient_.block<3, 3>(0, 0).setIdentity ();
+  point_jacobian_.setZero ();
+  point_jacobian_.block<3, 3>(0, 0).setIdentity ();
   point_hessian_.setZero ();
 
   Eigen::Transform<float, 3, Eigen::Affine, Eigen::ColMajor> eig_transformation;
   eig_transformation.matrix () = final_transformation_;
 
   // Convert initial guess matrix to 6 element transformation vector
-  Eigen::Matrix<double, 6, 1> p, delta_p, score_gradient;
+  Eigen::Matrix<double, 6, 1> transform, score_gradient;
   Eigen::Vector3f init_translation = eig_transformation.translation ();
   Eigen::Vector3f init_rotation = eig_transformation.rotation ().eulerAngles (0, 1, 2);
-  p << init_translation (0), init_translation (1), init_translation (2),
-  init_rotation (0), init_rotation (1), init_rotation (2);
+  transform << init_translation.cast<double>(), init_rotation.cast<double>();
 
   Eigen::Matrix<double, 6, 6> hessian;
 
-  double score = 0;
-
   // Calculate derivates of initial transform vector, subsequent derivative calculations are done in the step length determination.
-  score = computeDerivatives (score_gradient, hessian, output, p);
+  double score = computeDerivatives (score_gradient, hessian, output, transform);
```

Comment:
`const`

Reply:
```score``` cannot be const because it's fed to ```computeStepLengthMT``` as in/out variable at L137.